### DISCUSSION TITLE
Build wrong URL on signIn by apiBaseUrl

### DISCUSSION THREAD

**Discussion by maxvoronov** [nextauthjs/next-auth#9285] [Help] [answered]
Hi! I faced with a strange error and I'm not that it is a bug. Therefore I decided start the discussion, maybe I missed something.

So, I made simple custom credential provider:
```js
    providers: [
        CredentialsProvider({
            type: "credentials",
            async authorize(credentials, req) {
                const data = await authByCredentials(credentials) // Return profile and tokens
                return {
                    id: data.profile.id,
                    name: data.profile.name,
                    accessToken: data.token.access_token,
                    refreshToken: data.token.refresh_token,
                } as User;
            }
        }),
    ],
    pages: {signIn: "/signin"},
    ...
```

After I called signIn method:
```js
signIn("credentials", {verifyCode: verifyCode, redirect: false}).
    then((result) => {
        // Successfully authorized
        router.push('/');
        router.refresh();
    }).catch((err) => {
        // Something went wrong
        console.error("SignInForm error:", err)
    })
```

... and received error:
```
TypeError: URL constructor: /api/auth is not a valid URL
```

Debugging shown me that we got error when build URL because passed wrong one of args ([link to URL docs](https://developer.mozilla.org/en-US/docs/Web/API/URL/URL#url)):
https://github.com/nextauthjs/next-auth/blob/656c0cd81732e796e3053c8c83e7c7716fa94b09/packages/next-auth/src/react.tsx#L279
```js
// data.url = http://localhost:3000/signin
// baseUrl = /api/auth
const error = new URL(data.url, baseUrl).searchParams.get("error")
```

Perhaps it is worth adding an extra check here. What do you think? Or what did I miss?
Thanks in advance.

**Comment by tntchn** [comment, marked as answer]
https://github.com/nextauthjs/next-auth/pull/9286 found this was fixed in this pr

**Reply by maxvoronov** [reply]
Cool, thank you
